test/foundry/src/concrete/vault/OffchainAssetReceiptVault.tiers.t.sol

```diff
@@ -0,0 +1,218 @@
+        vm.assume(alice != bob);
+
+        fuzzedMinTier = uint8(bound(fuzzedMinTier, uint256(1), uint256(8)));
+
+        // Create the vault
+        OffchainAssetReceiptVault vault = createVault(alice, assetName, assetSymbol);
+
+        // Prank as Alice to grant roles
+        vm.startPrank(alice);
+
+        // Grant the necessary role
+        vault.grantRole(vault.ERC1155TIERER(), bob);
+        vm.stopPrank();
+
+        // Prank as Bob
+        vm.startPrank(bob);
+
+        // Emit the expected event
+        vm.expectEmit(true, true, true, true);
+        emit SetERC1155Tier(bob, tier, fuzzedMinTier, fuzzedContext, fuzzedData);
+
+        // Call the function that should emit the event
+        vault.setERC1155Tier(tier, fuzzedMinTier, fuzzedContext, fuzzedData);
+
+        // Stop the prank
+        vm.stopPrank();
+    }
+
+    /// Test authorizeReceiptTransfer reverts if unauthorizedSenderTier
+    function testTransferOnUnauthorizedSenderTier(
+        uint256 fuzzedKeyAlice,
+        uint256 fuzzedKeyBob,
+        string memory assetName,
+        bytes memory fuzzedData,
+        uint8 fuzzedMinTier,
+        uint256[] memory fuzzedContext,
+        uint256 certifyUntil,
+        uint256 referenceBlockNumber,
+        address tierAddress,
+        bool forceUntil
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address alice = vm.addr((fuzzedKeyAlice % (SECP256K1_ORDER - 1)) + 1);
+        address bob = vm.addr((fuzzedKeyBob % (SECP256K1_ORDER - 1)) + 1);
+
+        referenceBlockNumber = bound(referenceBlockNumber, 1, block.number);
+        certifyUntil = bound(certifyUntil, 1, type(uint32).max);
+
+        vm.assume(alice != bob);
+        vm.assume(tierAddress != address(0));
+
+        fuzzedMinTier = uint8(bound(fuzzedMinTier, uint256(1), uint256(8)));
+
+        // Create the vault
+        OffchainAssetReceiptVault vault = createVault(alice, assetName, assetName);
+
+        // Prank as Alice to grant roles
+        vm.startPrank(alice);
+
+        vault.grantRole(vault.CERTIFIER(), bob);
+        vault.grantRole(vault.ERC1155TIERER(), bob);
+
+        vm.stopPrank();
+
+        // Prank as Bob
+        vm.startPrank(bob);
+
+        // Call the certify function
+        vault.certify(certifyUntil, referenceBlockNumber, forceUntil, fuzzedData);
+
+        vault.setERC1155Tier(tierAddress, fuzzedMinTier, fuzzedContext, fuzzedData);
+
+        {
+            ITierV2 tierContract = ITierV2(tierAddress);
+            vm.mockCall(
+                address(tierContract),
+                abi.encodeWithSelector(ITierV2.reportTimeForTier.selector, bob, fuzzedMinTier, fuzzedContext),
+                abi.encode(999)
+            );
+
+            //Expect the revert with the exact revert reason
+            vm.expectRevert(abi.encodeWithSelector(UnauthorizedSenderTier.selector, bob, 999));
+
+            vault.authorizeReceiptTransfer(bob, alice);
```

Comment:
this checks the vault authorization logic, which we do need, but we also need to show that this authorization logic is used by the receipt when transfers are attempted